File: internal/domain/household.go
package domain

func BuildHouseholdSummary(
	finance FinanceSummary,
	commerce CommerceSummary,
) HouseholdSummary {
	financeOwners := mapFinanceOwners(finance.OwnerBreakdown)
	commerceOwners := mapCommerceOwners(commerce.OwnerBreakdown)
	return HouseholdSummary{Scopes: []HouseholdScopeSummary{
		ownerScopeSummary(
			"user",
			"User",
			financeOwners["user"],
			commerceOwners["user"],
		),
		ownerScopeSummary(
			"spouse",
			"Spouse",
			financeOwners["spouse"],
			commerceOwners["spouse"],
		),
		householdScopeSummary(finance, commerce),
	}}
}

func householdScopeSummary(
	finance FinanceSummary,
	commerce CommerceSummary,
) HouseholdScopeSummary {
	return HouseholdScopeSummary{
		Scope:                   "household",
		Label:                   "Household",
		Currency:                firstCurrency(finance.Currency, commerce.Currency, "KRW"),
		FinanceRecords:          finance.Records,
		FinanceCreditMinorUnits: finance.CreditMinorUnits,
		FinanceDebitMinorUnits:  finance.DebitMinorUnits,
		FinanceNetMinorUnits:    finance.NetMinorUnits,
		PurchaseRecords:         commerce.Records,
		PurchaseSpendMinorUnits: commerce.TotalSpendMinorUnits,
	}
}

File: apps/flutter/lib/snapshot/household.dart
part of '../snapshot.dart';

@immutable
class HouseholdScope {
  const HouseholdScope({
    required this.scope,
    required this.label,
    required this.currency,
    required this.financeRecords,
    required this.financeNetMinorUnits,
    required this.purchaseRecords,
    required this.purchaseSpendMinorUnits,
  });

  final String scope;
  final String label;
  final String currency;
  final int financeRecords;
  final int financeNetMinorUnits;
  final int purchaseRecords;
  final int purchaseSpendMinorUnits;
}
